ci: 'Could NOT find OpenSSL' fails

A month after 943a914664a3d18fc9bf897dd9bbba7b7f60b67f CI on Mac OS becomes broken again. Now it looks so:

```
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR) 
CMake Error at CMakeLists.txt:315 (message):
  Could NOT find OpenSSL development files (libssl-dev/openssl-devel package)
```

The problem was worked around in tarantool in 1.10.11-8-g321c093cd, 2.8.2-18-gc4b649bba and 2.10.0-beta1-65-g338309788, see https://github.com/tarantool/tarantool/issues/6468. It seems, we should apply the same patch for affected tarantool versions in our CI.

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -198,7 +198,7 @@ jobs:
         if: matrix.tarantool != 'brew' && matrix.tarantool != 'master'
 
       - name: Install tarantool build dependencies
-        run: brew install autoconf automake libtool
+        run: brew install autoconf automake libtool openssl@1.1
         if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
 
       - name: Clone tarantool ${{ env.T_VERSION }}
@@ -223,7 +223,11 @@ jobs:
           # There are tarantool releases on which AppleClang
           # complains about the problem that was fixed later in
           # https://github.com/tarantool/tarantool/commit/7e8688ff8885cc7813d12225e03694eb8886de29
-          cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo
+          #
+          # Set OpenSSL root directory for linking tarantool with OpenSSl of version 1.1
+          # This is related to #49. There are too much deprecations which affect the build and tests.
+          # Must be revisited after fixing https://github.com/tarantool/tarantool/issues/6477
+          cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1 -DOPENSSL_LIBRARIES=/usr/local/opt/openssl@1.1/lib
 
           # {{{ Workaround Mac OS build failure (gh-6076)
           #

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -214,6 +214,13 @@ jobs:
           fetch-depth: 0
         if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
 
+      - name: Patching tarantool for successful build
+        run: |
+          # These steps fix the problem with tarantool build described in
+          # https://github.com/tarantool/tarantool/issues/6576
+          git show 11e87877df9001a4972019328592d79d55d1bb01 | patch -p1
+        if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
+
       - name: Build tarantool ${{ env.T_VERSION }} from sources
         run: |
           mkdir "${T_DESTDIR}"
@@ -224,7 +231,7 @@ jobs:
           # complains about the problem that was fixed later in
           # https://github.com/tarantool/tarantool/commit/7e8688ff8885cc7813d12225e03694eb8886de29
           #
-          # Set OpenSSL root directory for linking tarantool with OpenSSl of version 1.1
+          # Set OpenSSL root directory for linking tarantool with OpenSSL of version 1.1
           # This is related to #49. There are too much deprecations which affect the build and tests.
           # Must be revisited after fixing https://github.com/tarantool/tarantool/issues/6477
           cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1 -DOPENSSL_LIBRARIES=/usr/local/opt/openssl@1.1/lib

